test/src/test/java/org/corfudb/integration/LogReplicationDynamicStreamIT.java

```diff
@@ -0,0 +1,386 @@
+                Metadata.class,
+                TableOptions.fromProtoSchema(IntValueTag.class)
+        );
+    }
+
+    public void writeToMap(Table<StringKey, IntValue, Metadata> map, boolean isActive,
+                           int startIndex, int totalEntries) {
+        int maxIndex = totalEntries + startIndex;
+        CorfuStore corfuStore = isActive ? corfuStoreActive : corfuStoreStandby;
+        for (int i = startIndex; i < maxIndex; i++) {
+            try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
+                txn.putRecord(map, StringKey.newBuilder().setKey(String.valueOf(i)).build(),
+                        IntValue.newBuilder().setValue(i).build(), null);
+                txn.commit();
+            }
+        }
+    }
+
+    public void verifyDataOnStandby(Table<StringKey, IntValue, Metadata> mapStandby,
+                                    int expectNumEntries) {
+        // Wait until data is fully replicated
+        while (mapStandby.count() != expectNumEntries) {
+            log.trace("Current map size on Sink:: {}", mapStandby.count());
+            // Block until expected number of entries is reached
+        }
+
+        // Verify data is present in Sink Site
+        assertThat(mapStandby.count()).isEqualTo(expectNumEntries);
+
+        try (TxnContext txn = corfuStoreStandby.txn(CORFU_SYSTEM_NAMESPACE)) {
+            for (int i = 0; i < expectNumEntries; i++) {
+                StringKey key = StringKey.newBuilder().setKey(String.valueOf(i)).build();
+                CorfuStoreEntry<StringKey, IntValue, Metadata> entry = txn.getRecord(mapStandby, key);
+                assertThat(entry.getPayload()).isNotNull();
+                assertThat(entry.getPayload()).isEqualTo(IntValue.newBuilder().setValue(i).build());
+            }
+            txn.commit();
+        }
+    }
+
+    /*
+     * Helper methods section end
+     */
+
+    /**
+     * Note that we already have basic end-to-end test for snapshot sync
+     * and log entry sync. {@link CorfuReplicationE2EIT} This test here
+     * is mainly for validating their behavior when tables / streams are
+     * only opened at Source side.
+     *
+     * (1) Set up Source and Sink CorfuRuntime and CorfuStore
+     * (2) Open mapA only at Source, write some entries to it
+     * (3) Start log replication server for snapshot sync
+     * (4) Open mapA at Sink and verify data replicated successfully
+     * (5) Write more entries to mapA at Source and verify log entry sync at Sink
+     */
+    @Test
+    public void testSnapshotAndLogEntrySync() throws Exception {
+        // Open mapA on Source
+        openMapAOnActive();
+
+        // writeToSource for initial snapshot sync
+        writeToMap(mapA, true, 0, numWrites);
+
+        // Confirm data does exist on Source Cluster
+        assertThat(mapA.count()).isEqualTo(numWrites);
+
+        startLogReplicatorServers();
+
+        // Open mapA on Sink after log replication started
```

Comment:
based on the description of this test, mapA should not be opened before snapshot sync completes, right?